### PR TITLE
Devfile to develop VS Code extensions

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?
Adds a devfile to develop VS Code extensions in Che-Theia 

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/14228

### CODE REVIEW

File: devfiles/vscode-extensions.devfile.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+projects:
+
+  - name: che-theia
+    source:
+      type: git
+      location: 'https://github.com/eclipse/che-theia.git'
+
+components:
+
+  - id: eclipse/che-theia/next
```
Comment (olexii4) [inline devfiles/vscode-extensions.devfile.yaml]: I do not agree with using the 'next' tag. Let us remove 'cheEditor' and will be using the default(which depends on infrastructure)

File: devfiles/vscode-extensions.devfile.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    actions:
+      - workdir: /projects/helloworld-vscode
+        type: exec
+        command: >
+          npm install;
+          yes | vsce package;
+          echo -e "\e[32mDone.\e[0m Packaging complete"
+        component: che-dev
+
+  - name: 3. run ... HOSTED Che-Theia + Hello World extension
```
Comment (sunix) [inline devfiles/vscode-extensions.devfile.yaml]: what if the vscode extension needs to be tested in a sidecar container ? (like most of the vscode extension we have)

File: devfiles/vscode-extensions.devfile.yaml (hunk 1 above)
Comment (vitaliy-guliy) [reply devfiles/vscode-extensions.devfile.yaml]: If we remove `cheEditor` at all, we will not be able to run commands in Theia editor container.

Comment (vitaliy-guliy) [reply devfiles/vscode-extensions.devfile.yaml]: Just to compare, you can take a look at the devfile to develop Che-Theia https://github.com/eclipse/che-theia/blob/master/devfiles/che-theia-all.devfile.yaml

File: devfiles/vscode-extensions.devfile.yaml (hunk 2 above)
Comment (vitaliy-guliy) [reply devfiles/vscode-extensions.devfile.yaml]: I think it's possible. Let's create a dedicated PR with this improvement.

**Comment by vitaliy-guliy** [comment]
@SkorikSergey this PR should not affect the build and tests. Is it possible to merge?

**Comment by SkorikSergey** [comment]
> @SkorikSergey this PR should not affect the build and tests. Is it possible to merge?

Yes. HappyPath test fails according to https://github.com/eclipse/che/issues/15586.

**Comment by vitaliy-guliy** [comment]
ci-build

**Comment by vitaliy-guliy** [comment]
ci-build

**Comment by RomanNikitenko** [comment]
crw-ci-test

**Comment by RomanNikitenko** [comment]
@vitaliy-guliy 
sorry for my comment above, I would like to test  - are happy-path-tests working in general?
because for my PR they are still broken...

**Comment by SkorikSergey** [comment]
crw-ci-test

File: devfiles/vscode-extensions.devfile.yaml (hunk 2 above)
Comment (sunix) [reply devfiles/vscode-extensions.devfile.yaml]: this devfile is doing it: https://github.com/sunix/devfiles/blob/480188119d223d631d17472621e0cef618e22dc9/devfiles/vscode-quarkus/devfile.yaml

**Comment by dmytro-ndp** [comment]
crw-ci-test

File: devfiles/vscode-extensions.devfile.yaml (hunk 2 above)
Comment (vitaliy-guliy) [reply devfiles/vscode-extensions.devfile.yaml]: I will create another PR with this command.
There is a big difference in use cases, that the workspace doesn't have a special sidecar container.
But I can add a command to launch Che-Theia in `theia-ide` container and to launch VS Code extension in `che-dev` container.

**Pull request merged**